credentials/src/test/kotlin/web5/sdk/credentials/VerifiableCredentialTest.kt

```diff
@@ -94,40 +101,75 @@ class VerifiableCredentialTest {
+  fun createJwk() {
+    val privateJwk = JWK.parse(
+      """{
+          "kty": "EC",
+          "d": "OeBCt3M8roz9F9Ny192TtZXf_-qPzAkwSM6ep7QdSO4",
+          "use": "sig",
+          "crv": "secp256k1",
+          "kid": "NB_qlUIr06-AikVVMFreq0lc-omQtzc6lwhhcvgO6r4",
+          "x": "DdtN8W6x_34pB_nkxR0e1tmDkNnsJeusBAEPzKWgf_Y",
+          "y": "u3W135inodLqtcEb9jNGS3JsM_uFKmkJSb8Trc9luWI",
+          "alg": "ES256K"
+        }"""
     )
+    val publicKey = privateJwk.toPublicJWK()
+    val base64Encoded = Convert(publicKey.toJSONString()).toBase64Url(padding = false)
 
-    val vcJwt = vc.sign(issuerDid)
-    val parsedJwt = SignedJWT.parse(vcJwt) // validates JWT
+    val did = "did:jwk:$base64Encoded"
+    println(did)
+  }
 
-    val didDocument = issuerDid.resolve().didDocument
-    val assertionMethod = didDocument.assertionMethodVerificationMethodsDereferenced.first()
+  data class TestVectors(
+    val description: String,
+    val vectors: List<TestVector>
+  )
 
-    val jwtHeader = parsedJwt.header
-    assertNotNull(jwtHeader.algorithm)
-    assertEquals(JOSEObjectType.JWT, jwtHeader.type)
-    assertEquals(assertionMethod.id.toString(), jwtHeader.keyID)
+  data class TestVector(
+    val description: String,
+    val input: TestInput,
+    val output: String?,
+    val errors: Boolean?,
+  )
 
-    val jwtClaims = parsedJwt.jwtClaimsSet
-    assertNotNull(jwtClaims.issueTime)
-    assertNotNull(jwtClaims.getClaim("vc"))
-    assertEquals(issuerDid.uri, jwtClaims.issuer)
-    assertEquals(holderDid.uri, jwtClaims.subject)
+  data class TestInput(
+    val signerDidUri: String?,
+    val signerPrivateJwk: Map<String, Any>?,
+    val credential: Map<String, Any>?,
+  )
 
-    val vcDataModelJson = jwtClaims.getJSONObjectClaim("vc")
-    val vcDataModel = VcDataModel.fromMap(vcDataModelJson)
+  @Test
+  fun `creates test vectors`() {
+    // read a map from a json file
+    val mapper = jacksonObjectMapper()
+    val testVectors =
+      mapper.readValue(File("../test-vectors/credentials/create.json"), TestVectors::class.java)
+
+    testVectors.vectors.filterNot { it.errors ?: false }.forEach { vector ->
+      val testInput = vector.input
+
+      val vc = VerifiableCredential.fromJson(mapper.writeValueAsString(testInput.credential))
+
+      val keyManager = InMemoryKeyManager()
+      keyManager.import(listOf(testInput.signerPrivateJwk!!))
+      val issuerDid = Did.load(testInput.signerDidUri!!, keyManager)
+      val vcJwt = vc.sign(issuerDid)
+
+      assertEquals(vector.output, vcJwt, vector.description)
+    }
 
-    assertEquals(holderDid.uri, vcDataModel.credentialSubject.id.toString())
-    assertContains(vcDataModel.types, "StreetCred")
+    testVectors.vectors.filter { it.errors ?: false }.forEach { vector ->
+      val testInput = vector.input
+      assertFails(vector.description) {
+        VerifiableCredential.fromJson(mapper.writeValueAsString(testInput.credential))
+      }
+    }
```

Comment:
Just a thought on structure here:

We could have two test vectors: `create-success.json` and `create-failure.json`. That way we could have separate functions that test success/failure scenario scenarios instead of filtering on the inputs. 

But I'm ok with this too! I'm sure our test vectors will evolve over time, so definitely not blocking by any means.

Reply:
I like that! Used `_` instead of `-` because it's what was allowed :)